index.js

```diff
@@ -1,59 +1,103 @@
 'use strict';
 
 const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor'; // Key in annotations object that maps to an executor NPM module
+const SCHEMA_START = require('screwdriver-data-schema').plugins.queue.start;
 const Executor = require('screwdriver-executor-base');
+const Joi = require('joi');
+const Resque = require('node-resque');
+
+/**
+ * Strip the leading 'screwdriver-executor' from plugin names
+ * @method getPluginName
+ * @param {String} plugin
+ * @return {String}
+ */
+function getPluginName(plugin) {
+    return plugin.replace('screwdriver-executor-', '');
+}
 
-class ExecutorRouter extends Executor {
+class ExecutorQueue extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
      * @param  {Object}         config                      Object with executor and ecosystem
      * @param  {String}         [config.defaultPlugin]      Optional default executor
-     * @param  {Object}         [config.ecosystem]          Optional object with ecosystem values
+     * @param  {Object}         config.ecosystem            Object with ecosystem values
      * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {Object}         config.redisConnection      Connection details for redis
      * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
      * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
         const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
```

Comment:
The jsdoc says that config.executor can be a single executor object. In that case doesn't this error get thrown if it is a single executor object based on the `!(Array.isArray(executorConfig))` statement?